code/modules/economy/cargo_telepads.dm

```diff
@@ -0,0 +1,28 @@
+/obj/machinery/telepad_cargo
+	name = "cargo telepad"
+	desc = "A telepad used to recieve imports and send exports."
+	icon = 'icons/obj/machines/telepad.dmi'
+	icon_state = "pad-idle"
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 20
+	active_power_usage = 500
+	var/telepad_id = 0
+/obj/machinery/telepad_cargo/New()
+	telepad_id = random_id(type,10000,99999)
+
+/obj/machinery/telepad_cargo/attackby(obj/item/O as obj, mob/user as mob, params)
+	if(IS_WRENCH(O))
```

Comment:
There's a base class handling for wrenching/unwrenching from the floor. You can just set the flag ``OBJ_FLAG_ANCHORABLE`` in ``obj_flags``. The base class will handle wrenching/unwrenching in it's attackby proc.